File: football-collection-builder/backend/app/models/catalog_quality.py
from typing import Any
from pydantic import BaseModel, Field
class CatalogQualityRequest(BaseModel): replacePrevious: bool=True
class CatalogQualityRun(BaseModel):
    id:int|None=None; catalogBuildRunId:int; startedAt:str; finishedAt:str|None=None; durationMs:int=0; status:str
    totalIssues:int=0; autoResolved:int=0; reviewRequired:int=0; qualityScore:float=0; message:str
class CatalogQualityStatus(BaseModel):
    catalogAvailable:bool; qualityAnalysisAvailable:bool; catalogBuildRunId:int|None=None; lastAnalysis:CatalogQualityRun|None=None; status:str
class CatalogQualitySummary(BaseModel):
    totalIssues:int;openIssues:int;autoResolvedIssues:int;reviewRequiredIssues:int;issuesByType:dict[str,int];issuesBySeverity:dict[str,int]
    countriesTotal:int;countriesConfirmed:int;countriesInferred:int;countriesUnknown:int;teamsTotal:int;teamsConfirmed:int;teamsInferred:int;teamsUnknown:int
    itemsTotal:int;itemsConfirmed:int;itemsInferred:int;itemsUnknown:int;collectionsTotal:int;collectionsClassified:int;collectionsUnknown:int;missingImages:int
    qualityScore:float;analyzedAt:str;durationMs:int
class QualityPage(BaseModel): items:list[dict[str,Any]]=Field(default_factory=list);total:int;limit:int;offset:int

File: football-collection-builder/frontend/src/types/catalogQuality.ts
export type QualityRun={id:number;catalogBuildRunId:number;finishedAt:string|null;durationMs:number;status:string;totalIssues:number;autoResolved:number;reviewRequired:number;qualityScore:number}
export type QualityStatus={catalogAvailable:boolean;qualityAnalysisAvailable:boolean;catalogBuildRunId:number|null;lastAnalysis:QualityRun|null;status:string}
export type QualitySummary={totalIssues:number;openIssues:number;autoResolvedIssues:number;reviewRequiredIssues:number;issuesByType:Record<string,number>;issuesBySeverity:Record<string,number>;countriesUnknown:number;teamsUnknown:number;collectionsUnknown:number;missingImages:number;qualityScore:number;analyzedAt:string;durationMs:number}
export type QualityIssue={id:number;issueType:string;severity:string;entityType:string;relativePath:string;message:string;resolutionStatus:string;pattern:string;evidence:string;reason:string;ruleCode:string|null;confidence:string|null}
export type Resolution={id:number;issueId:number;ruleCode:string;previousValue:string|null;resolvedValue:string|null;confidence:string;reason:string}
export type QualityGroup={key:string;issueType:string;pattern:string;count:number;auto_resolvable_count:number;review_required_count:number}
export type Page<T>={items:T[];total:number;limit:number;offset:number}
